doc/page-properties.h

```diff
@@ -0,0 +1,154 @@
+**
+** SPDX-License-Identifier: CC-BY-SA-4.0
+**
+**********************************************************************/
+
+/**
+\page page-properties Properties
+
+\section properties-intro Introduction
+Interface definition files (*.qface) can define properties. See https://github.com/Pelagicore/qface for information
+about QFace interface definition.
+
+In Facelift each property is exposed to the QML engine in a generated adapter class, as Q_PROPERTY, with setter
+and getter functions, and a change signal. If the property is marked as 'readonly' then no setter is generated.
+The generated class \<Inerface name\>QMLAdapter wrapps the generated service, which is the owner of the properties.
+
+In Facelift properties are generated as members of the \<Inerface name\>ImplementationBase C++ class.
+The facelift::Property class template provides encapsulation for the internal data. To get the underlying value
+developer can use a getter that returns the const reference to the underlying data (in case of the service property
+it is const reference to a pointer) or one of the specialization specific access functions. The value can be changed only by
+using a setter, operator= or one of the specialization specific access functions. There is also a set of change operators
+provided: ++, \--, -=, +=, *= and /=. This guarantees that the data will not be unnoticibly changed. Each time the setter
+with a value different than the previous one, or other operation that changes the value is called, the change signal is emitted.
+
+\section QML exposure
+
+In case of C++ interface implementation, the property, genetated as a C++ code, is exposed directly to the QML engine
+by Q_PROPERTY macro in the relevant QMLAdapter and can be used without any additional effort.
+
+\section cpp C++
+
+Each property generated as C++ code is encapsulated inside one of the templated property containers: Property<T>, ListProperty<T>,
+ServiceProperty<T> or ModelProperty<T>. Additionally ListProperty and QMap<QString, T> partial template specializations
+add extra convenience functions: void removeAt(int i), void addElement(ElementType element) and int size() const.
+The QMap<QString, T> specialization is created for the map property defined in *.qface file.
+
+\section subservice Property ownership
+
+The service owns the property objects and maintains their life cycle.
+
+The service property stores a pointer to another service. In this case property object is not an owner of the
+pointed service. In terms of QML engine, the service object ownership is held by C++ code.
+See https://doc.qt.io/qt-5/qqmlengine.html#ObjectOwnership-enum
+
+Other property types are owners of the underyling objects.
+
+
+\section bindings Property bindings
+
+Typically property state change may happen on setter call. An alternative way of updating property value with a minimal
+effort is to bind it to the signal. Whenever property is notified (state change triggered), the property will call
```

Comment:
The "bind" mechanism should be introduced in the same section as the other mechanism (using setValue(), =operator or other means).
Please use articles.